pandas-stubs/core/indexing.pyi

```diff
@@ -83,6 +83,18 @@ class _LocIndexerFrame:
         value: Union[DataFrame, Series, float],
     ) -> None: ...
 
+class _AtIndexerFrame:
+    # get item
+    def __getitem__(self, idx: Tuple[int, _StrLike]) -> float: ...
```

Comment:
you were concerned about returning just `float` here, right? It's a big problem with DataFrame. I spent a lot of time thinking about it, but couldn't find a perfect solution. The problem is that a DataFrame can have multiple types. Each column can have a different type.

_Ideally_, we would make DataFrame generic with _all_ the column types. Something like:

```python
df: DataFrame[["column_a", int], ["column_b", float]] = pd.DataFrame(
    [[3, 0.3], [4, 0.4]], columns=["column_a", "column_b"]
)
x: int = df["column_a"].iloc[0]
y: float = df["column_b"].iloc[1]
```

But this would make the type stubs *super complicated*, if it is even possible.

So, for now, I think we cannot get around returning a `Union` here:

```suggestion
    def __getitem__(self, idx: Tuple[int, _StrLike]) -> Union[int, float, str]: ...
```

Returning `Union` always sucks because then you don't know what the result is, but I don't see a better solution.